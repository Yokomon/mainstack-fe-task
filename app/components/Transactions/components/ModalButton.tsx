"use client";

interface ModalButtonProps {
  text: string;
}

const ModalButton: React.FC<ModalButtonProps> = ({ text }) => {
  return (
    <button className="p-3 flex items-center px-4 font-medium space-x-3 rounded-full hover:bg-gray-50 ring-1 ring-gray-200 shadow-sm duration-500">
      <h3 className="text-xs">{text}</h3>
    </button>
  );
};

export default ModalButton;
