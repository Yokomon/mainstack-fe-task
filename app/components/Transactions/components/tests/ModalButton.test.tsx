import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

import ModalButton from "../ModalButton";

describe("ModalButton Component", () => {
  it("renders without crashing", () => {
    render(<ModalButton text="Test Button" />);
    expect(screen.getByText("Test Button")).toBeInTheDocument();
  });

  it("matches snapshot", () => {
    const { asFragment } = render(<ModalButton text="Test Button" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
