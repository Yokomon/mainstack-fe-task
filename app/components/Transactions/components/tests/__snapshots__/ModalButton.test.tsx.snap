// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalButton Component matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="p-3 flex items-center px-5 space-x-2 rounded-full hover:bg-gray-50 ring-1 ring-gray-200 shadow-sm duration-500"
  >
    <h3
      class="text-xs"
    >
      Test Button
    </h3>
  </button>
</DocumentFragment>
`;
